### PR TITLE
Add type check handling for php8 array functions

### PR DESCRIPTION
**Pull request by mchanDev**
## Description

This adds some error handling before using array functions from PHP core to prevent type warnings from becoming fatals

### Plugin Updated: Role Backup

Add type check handling for php8 array functions for role backup CLI function

## Pre-review checklist

Please make sure the items below have been covered before requesting a review:

- [X] This change works and has been tested locally (or has an appropriate fallback).
- [X] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- N/A This change uses a rollout method to ease with deployment (if applicable - especially for large scale actions that require writes).
- N/A This change has relevant documentation additions / updates (if applicable).
- [X] I've created a changelog description that aligns with the provided examples.

## Pre-deploy checklist

- [ ] VIP staff: Ensure any alerts added/updated conform to internal standards (see internal documentation). 

## Steps to Test

Begin by backing up current roles, then alter roles and backup again:

wp cron event run 'Automattic\VIP-backup_roles'
wp role delete editor
wp role delete author
wp role delete contributor
wp cron event run 'Automattic\VIP-backup_roles'

A couple options will now be listed in wp vip role-backup list

    view all backups: wp vip role-backup list
    view a backup: wp vip role-backup view [<time_key>]
    and restore: wp vip role-backup restore [<time_key>]
    and confirm: wp role list

This ensures no functionality has changed as is.

You can then remove the `vip_backup_user_roles` option:

wp_option_delete vip_backup_user_roles
wp vip role-backup list

This should result in the message `The option vip_backup_user_roles is not an array`

-----------

To test the change to detect $role['capabilites'], adding a value override `$role['capabilites'] = null` at line 268 is easiest to test this change.

### CODE REVIEW

File: vip-helpers/class-vip-backup-user-role-cli.php [hunk 1]
```diff
@@ -28,6 +28,12 @@ class VIP_Backup_User_Role_CLI extends \WPCOM_VIP_CLI_Command {
 	public function list_( $args, $assoc_args ) {
 
 		$backup_roles = get_option( 'vip_backup_user_roles', [] );
+		if ( ! is_array( $backup_roles ) ) {
+			\WP_CLI::warning( 'The option vip_backup_user_roles is not an array' );
+			$debug_export = var_export( $backup_roles, true );
+			\WP_CLI::error_multi_line( $debug_export );
+			\WP_CLI::error( 'exit as nothing to do' );
```
Comment (rinatkhaziev) [inline vip-helpers/class-vip-backup-user-role-cli.php]: This line and similar below seem a bit redundant, I think silently bailing is enough?

File: vip-helpers/class-vip-backup-user-role-cli.php [hunk 2]
```diff
@@ -28,6 +28,12 @@ class VIP_Backup_User_Role_CLI extends \WPCOM_VIP_CLI_Command {
 	public function list_( $args, $assoc_args ) {
 
 		$backup_roles = get_option( 'vip_backup_user_roles', [] );
+		if ( ! is_array( $backup_roles ) ) {
+			\WP_CLI::warning( 'The option vip_backup_user_roles is not an array' );
```
Comment (rinatkhaziev) [inline vip-helpers/class-vip-backup-user-role-cli.php]:
```suggestion
			\WP_CLI::warning( 'The option vip_backup_user_roles is not an array, aborting' );
```

**Review by rinatkhaziev — DISMISSED** [review]
@mchanDev thanks for the PR! Left some inline feedback.

File: vip-helpers/class-vip-backup-user-role-cli.php (hunk 1 above)
Comment (mchanDev) [reply vip-helpers/class-vip-backup-user-role-cli.php]: I can replace with an exit call

**Review by rinatkhaziev — APPROVED** [review]

**Pull request merged**